### PR TITLE
[#187364682] ft-Users should be presented with a reset password link 

### PR DESCRIPTION
**Pull request by Bananayosostene**
**What does this PR do?**
This PR introduces the "Forgot Password" functionality to the application. Users who have forgotten their passwords can now request a password reset link by providing their email address. This feature ensures users can regain access to their accounts securely and conveniently. The motivation for adding this feature is to enhance user experience by providing a method to recover account access without administrative intervention.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**
The feature has been tested locally by simulating user interactions with the "Forgot Password" and "Reset Password" forms.
Verified that emails are sent successfully and that the reset link works as expected.
Ensured that the new password is correctly updated and allows the user to log in with the new credentials.
Unit tests have been added to cover the new functionality.

Checklist:
 My code follows the style guidelines of this project
 I have performed a self-review of my own code
 I have commented my code, particularly in hard-to-understand areas
 My changes generate no new warnings
 I have added tests that prove my fix is effective or that my feature works
 New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/test/App.test.tsx [hunk 1]
```diff
@@ -62,10 +63,22 @@ describe("Hello Testing",()=>{
...
 })
 
+// describe("Should login", () => {
+//   it("should render Login ", () => {
+//     render(<Login />);
+//     const myElement = screen.getByTestId("test-id");
+//     expect(myElement).toBeInTheDocument();
+//   });
+// });
+
```
Comment (AngeloChristian1) [inline src/test/App.test.tsx]:
```suggestion
// describe("Should login", () => {
//   it("should render Login ", () => {
//     render(<Login />);
//     const myElement = screen.getByTestId("test-id");
//     expect(myElement).toBeInTheDocument();
//   });
// });

```

File: src/views/ResetPassword.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                  messageType === "success"
+                    ? "success-message"
+                    : "error-message"
+                }
+              >
+                {message}
+              </p>
+            )}
+          </div>
+          <button className="reset-btn text-sm" onClick={handleSubmit}>
```
Comment (AngeloChristian1) [inline src/views/ResetPassword.tsx]: You can put the handleSubmit function on the form tag

**Review by Furaha-Justine — APPROVED** [review]
Good job.

**Review by AngeloChristian1 — APPROVED** [review]

File: src/views/ForgotPassword.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      .required("Email is required")
+      .matches(emailPattern, "Please enter a valid email"),
+  });
+
+  const handleForgotPassword = async (values: { email: string }) => {
+    try {
+      const response = await axios.post(
+        "https://e-commerce-bitcrafters-bn-1mpf.onrender.com/users/reset/link",
+        { email: values.email },
+      );
```
Comment (munezerobagira) [inline src/views/ForgotPassword.tsx]:
```suggestion
  const axiosClient=useAxiosInstance()
  const handleForgotPassword = async (values: { email: string }) => {
    try {
      const response = await axiosClient.post(
        "/users/reset/link",
        { email: values.email },
      );
```
You can use the instance that @AngeloChristian1 have worked on

**Review by munezerobagira — CHANGES_REQUESTED** [review]

File: src/views/ForgotPassword.tsx (hunk 3 above)
Comment (Bananayosostene) [reply src/views/ForgotPassword.tsx]: Yes i can use it

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**